### ISSUE TITLE
Error using lockfiles with conanfile.txt

### ISSUE THREAD

**Issue by czoido** [conan-io/conan#5499]
Using: Windows and conan version 1.17.1

I get `ERROR: Unspecified reference in graph-lock, please specify` when trying to use lockfiles in a package that has `conanfile.txt` instead of `conanfile.py`

To reproduce:
```
git clone https://github.com/conan-io/examples.git
cd examples/libraries/poco/md5
mkdir build && cd build
conan install ..
conan install .. --lockfile
```
Output:
```
Using lockfile: 'C:\temp\examples\libraries\poco\md5\build\conan.lock'
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=15
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

ERROR: Unspecified reference in graph-lock, please specify
```

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [X] I've specified the Conan version, operating system version and any tool that can be relevant.
- [X] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by rbrich** [comment]
I have the same issue, is there a workaround other than switching to `conanfile.py`? I'm trying to lock dependencies for a project (which doesn't have Conan package by itself).

**Comment by lasote** [comment]
To be released at Conan 1.20
